spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisStreamInboundChannelAdapter.java

```diff
@@ -0,0 +1,171 @@
+
+	private final ReactiveRedisConnectionFactory reactiveConnectionFactory;
+
+	private final MessageConverter messageConverter = new SimpleMessageConverter();
+
+	private final String streamKey;
+
+	private StreamReceiver receiver;
+
+	private StreamReceiver.StreamReceiverOptions receiverOptions;
+
+	private volatile String consumerGroupName;
+
+	private volatile String consumerName;
+
+	private volatile boolean createGroupIfNotExist = false;
+
+	private  ReadOffset readOffset = ReadOffset.lastConsumed();
+
+	private RedisTemplate template;
+
+	public RedisStreamInboundChannelAdapter( String streamKey,
+			ReactiveRedisConnectionFactory reactiveConnectionFactory ) {
+		this.streamKey = streamKey;
+		this.reactiveConnectionFactory = reactiveConnectionFactory;
+	}
+
+	public void setConsumerGroupName( @Nullable String consumerGroupName ) {
+		this.consumerGroupName = consumerGroupName;
+	}
+
+	public void setConsumerName( @Nullable String consumerName ) {
+		this.consumerName = consumerName;
+	}
+
+	public void setCreateGroupIfNotExist( boolean createGroupIfNotExist ) {
+		this.createGroupIfNotExist = createGroupIfNotExist;
+	}
+
+	public void setReceiver( StreamReceiver receiver ) {
+		this.receiver = receiver;
+	}
+
+	public void setReceiverOptions( StreamReceiver.StreamReceiverOptions receiverOptions ) {
+		this.receiverOptions = receiverOptions;
+	}
+
+	public void setReadOffset( ReadOffset readOffset ) {
+		this.readOffset = readOffset;
+	}
+
+	void setTemplate( RedisTemplate template ) {
+		this.template = template;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		registerListener();
+	}
+
+	@Override
+	protected void doStart() {
+		super.doStart();
+		createGroup();
+	}
+	
+	@Override
+	public String getComponentType() {
+		return "redis:stream-inbound-channel-adapter";
+	}
+
+	private Message<?> convertMessage( Record<Object, Object> record ) {
+		Map<String, Object> headers = new HashMap<>();
+		headers.put( RedisHeaders.STREAM_KEY, record.getStream() );
+		headers.put( RedisHeaders.STREAM_MESSAGE_ID, record.getId() );
+		return this.messageConverter.toMessage( record.getValue(), new MessageHeaders( headers ) );
+	}
+
+	private void registerListener() {
+		if ( this.receiverOptions != null ) {
+			this.receiver = StreamReceiver.create( reactiveConnectionFactory, this.receiverOptions );
+		}
+		else {
+			this.receiver = StreamReceiver.create( reactiveConnectionFactory );
+		}
+
+		StreamOffset offset = StreamOffset.create( this.streamKey, this.readOffset );
+
+		if ( StringUtils.isEmpty( this.consumerGroupName ) ) {
+			this.receiver
+					.receive( offset )
+					.subscribe( record -> sendMessage( convertMessage( ( Record<Object, Object> ) record ) ) );
+		}
+		else {
+			Assert.hasText( consumerName, "'consumerName' must be set" );
+			Consumer consumer = Consumer.from( this.consumerGroupName, this.consumerName );
+			this.receiver
+					.receiveAutoAck( consumer, offset )
```

Comment:
Looks like we also need to think about manual acknowledgement:
```
 * Every record must be acknowledged using
	 * {@link org.springframework.data.redis.connection.ReactiveStreamCommands#xAck(ByteBuffer, String, String...)}
```

See `receive()` API for the `StreamReceiver`.
This has to be populated as a `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK`.
And also see `SimpleAcknowledgment`  abstraction to wrap the call to that `xAck`. for the particular record returned.

Saying that I'd expect an `autoAck(true|false)` option on this channel adapter.